invenio_rdm_records/services/components/pids.py

```diff
@@ -212,12 +212,18 @@ def delete_record(self, identity, data=None, record=None, uow=None):
             self.service.pids.parent_pid_manager.discard_all(
                 parent_pids, soft_delete=True
             )
+        else:
+            # We're sending a task in case there is a race condition with two
+            # versions being deleted at the same time to ensure that we have
+            # consistent database state
+            self.uow.register(TaskOp(cleanup_parent_pids, record["id"]))
 
         # Async register/update tasks after transaction commit.
         for scheme in parent_pids.keys():
-            self.uow.register(
-                TaskOp(register_or_update_pid, record["id"], scheme, parent=True)
-            )
+            if record:
```

Comment:
Looking at the code, this might not be possible. With the previous `if`, if record is None, it will fail with `record.parent` or `record["id"]` (in both condition's branches).

Reply:
yeah my bad this was completely incorrect